core/src/main/java/hudson/model/View.java

```diff
@@ -1006,6 +1009,36 @@ public synchronized void doDoDelete(StaplerRequest req, StaplerResponse rsp) thr
     public abstract Item doCreateItem( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException;
 
     /**
+     * Makes sure that the given name is good as a job name.
+     * For use from {@code newJob}.
+     */
+    @Restricted(DoNotUse.class) // called from newJob view
+    public FormValidation doCheckJobName(@QueryParameter String value) {
+        // this method can be used to check if a file exists anywhere in the file system,
+        // so it should be protected.
+        getOwner().checkPermission(Item.CREATE);
+
+        if (Util.fixEmpty(value) == null) {
+            return FormValidation.ok();
+        }
+
+        try {
+            Jenkins.checkGoodName(value);
```

Comment:
I have to include a similar validation in the client-side part, specifically, [here](https://github.com/jenkinsci/jenkins/pull/2191/files#diff-e146d4fdbfe6385a456aea9775f6282dR53).